### PR TITLE
Fix Windows User Mode Build Breaks from VS v16.10 Update

### PR DESCRIPTION
**Pull request by nibanks**
AZP just started pushing an update that includes the VS 2019 v16.10 update, causing many of our builds to fail. This PR fixes most of those issues so at least Windows user mode builds pass. Kernel mode still has a link error that I can't quite figure out.

### CODE REVIEW

**Review by anrossi — APPROVED** [review]

File: src/inc/quic_platform_winkernel.h [hunk 1]
```diff
@@ -30,6 +30,7 @@
 #pragma warning(disable:26071)
 #pragma warning(disable:28118)
 #pragma warning(disable:28196)
+#pragma warning(disable:28251)
```
Comment (anrossi) [inline src/inc/quic_platform_winkernel.h:33]: we probably shouldn't ignore this one, though. It tells us if our header/source annotations are in sync. Or it should.

Comment (nibanks) [reply src/inc/quic_platform_winkernel.h:33]: We're just ignoring this on public headers for the most part. We don't want to fail our build if the public SDK/WDK headers have issues (which is what is happening here).

**Comment by ThadHouse** [comment]
Im not surprised stddef.h fails in kernel mode. Can you try just copying the offsetof macro locally and seeing if that works. Still a bug, but there's no way to revert the builds on azp.

**Comment by nibanks** [comment]
> Im not surprised stddef.h fails in kernel mode. Can you try just copying the offsetof macro locally and seeing if that works. Still a bug, but there's no way to revert the builds on azp.

stddef.h works fine on Windows kernel. I'm just seeing the unresolved symbol error now for _vsnprintf_s.

**Pull request merged**